components/HistoryTable.qml

```diff
@@ -52,6 +54,22 @@ ListView {
             + translationManager.emptyString;
     }
 
+    function lookupPaymentID(paymentId) {
+        if (!addressBookModel)
+            return ""
+        if (paymentId.length == 16)
+            paymentId += "000000000000000000000000000000000000000000000000"
+        var count = addressBookModel.rowCount()
+        for (var i = 0; i < count; i++) {
+            var idx = addressBookModel.index(i, 0)
+            var ati = addressBookModel.data(idx, AddressBookModel.AddressBookPaymentIdRole)
+            if (ati.length == 16)
+                ati += "000000000000000000000000000000000000000000000000"
+            if (ati == paymentId)
+                return addressBookModel.data(idx, AddressBookModel.AddressBookDescriptionRole)
```

Comment:
Would prefer having the lookup functions in AddressBook instead. For later reuse.